### PR TITLE
add ACL support for health checks controller

### PR DESCRIPTION
**Pull request by kschoche**
Changes proposed in this PR:
- server-acl-init will now add additional rules for the connect-inject-webhook when ACLs and health checks are enabled.
- deprecate `-create-inject-auth-method`
- deprecate `-create-inject-namespace-token`. This flag was always passed when `-enable-namespaces` was true so now we just look at`-enable-namespaces` and `-create-inject-token`
- undeprecate `-create-inject-token`. This will always be passed by helm if connect inject is enabled.
- move all namespaced server-acl-init/command_test.go tests that used `-create-inject-namespace-token` from oss to ent as these are now invalid for oss (they always were but they got away with being in oss because they weren't passing `-enable-namespaces`
- tests to validate backwards compatibility are added and coverage for new flags.

How I've tested this PR:
unit tests + some manual testing of the command flags

How I expect reviewers to test this PR:
run the unit tests and try out the full stack by grabbing
https://github.com/hashicorp/consul-helm/pull/665

This table summarizes what helm would pass with versions <= 0.25.0 (old) and > 0.25.0 (new):

|     | Enable Namespaces | Enable Healthchecks | Flags Passed                                                                   | Auth Method | Token | Global Token |
|-----|-------------------|---------------------|--------------------------------------------------------------------------------|-------------|-------|--------------|
| Old |       FALSE       | N/A                 | -create-inject-auth-method                                                     | Yes         | No    | N/A          |
|     |        TRUE       | N/A                 | -create-inject-auth-method, -enable-namespaces, -create-inject-namespace-token | Yes         | Yes   | Yes          |
|     |                   |                     |                                                                                |             |       |              |
| New |       FALSE       |        FALSE        | -create-inject-token                                                           | Yes         | No    | N/A          |
|     |        TRUE       |        FALSE        | -create-inject-token, -enable-namespaces                                       | Yes         | Yes   | Yes          |
|     |        TRUE       |         TRUE        | -create-inject-token, -enable-namespaces, -enable-healthchecks                 | Yes         | Yes   | Yes          |
|     |       FALSE       |         TRUE        | -create-inject-token, -enable-healthchecks                                     | Yes         | Yes   | No           |

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added - this will be part of the full health checks changelog entry

### CODE REVIEW

File: subcommand/inject-connect/command_test.go
Comment (kschoche) [inline subcommand/inject-connect/command_test.go:120]: typo fix from the original merge.

File: subcommand/server-acl-init/command.go
Comment (kschoche) [inline subcommand/server-acl-init/command.go:191]: I'm thinking of changing this to something like `add-health-check-rules-to-inject`

File: subcommand/server-acl-init/command.go
Comment (kschoche) [inline subcommand/server-acl-init/command.go:466]: if this is renamed to something like "add health checks rules" I can add a comment stating that it will imply the creation of the inject token as well.

File: subcommand/server-acl-init/rules.go
Comment (kschoche) [inline subcommand/server-acl-init/rules.go:215]: I'm not 100% sure on the need for `namespace_prefix` for the non consul-ent case.
Once consul-ent comes into play I'm sure this will be required.
I'll do some more testing today to validate.

Comment (kschoche) [reply subcommand/server-acl-init/rules.go:215]: I've confirmed it is not necessary for now and will be removing it with a commit before we merge.

File: subcommand/server-acl-init/rules.go
Comment (kschoche) [inline subcommand/server-acl-init/rules.go:214]:
```suggestion
{{- if .EnableHealthChecks }}
```

File: subcommand/server-acl-init/rules.go
Comment (kschoche) [inline subcommand/server-acl-init/rules.go:18]:
```suggestion
	EnableHealthChecks bool
```

**Review by lkysow — APPROVED** [review]

File: subcommand/server-acl-init/command.go
Comment (ishustava) [reply subcommand/server-acl-init/command.go:134]: Should we add a similar note to `-enable-namespaces`?

**Review by ishustava — APPROVED** [review]
Looks great! I've added some comments and suggestions, but I don't think they are blocking the merge.

File: subcommand/server-acl-init/rules.go
Comment (kschoche) [reply subcommand/server-acl-init/rules.go:220]: Yeah, you're right its off by 2 spaces but bringing in namespaces will offset it as well!

Comment (kschoche) [reply subcommand/server-acl-init/rules.go:220]: speaking of which I forgot to add this into the namespace PR, I'll rebase that one once this one is merged!

File: subcommand/server-acl-init/command_ent_test.go
Comment (kschoche) [reply subcommand/server-acl-init/command_ent_test.go:672]: good catch! I've added it back

File: subcommand/server-acl-init/command_test.go
Comment (kschoche) [reply subcommand/server-acl-init/command_test.go:830]: this was leftover from a previous attempt to resolve some issues with `t.Parallel()` which turned out to be something else, I've set it back to `t`!
Thanks!

**Pull request merged**
